crates/language/src/language_settings.rs

```diff
@@ -27,6 +27,27 @@ impl<'a> Into<SettingsLocation<'a>> for &'a dyn File {
 /// Initializes the language settings.
 pub fn init(cx: &mut AppContext) {
     AllLanguageSettings::register(cx);
+    DownloadConfiguration::register(cx);
+}
+
+#[derive(Default, Serialize, Deserialize, Clone)]
```

Comment:
you don't really need two structs for this. You could just use the `DownloadConfiguration` one with `#[serde(default)]` as an attribute macro to ensure that if it's missing on deserialization it defaults to false. It makes the behaviour easier to maintain and more predictable rather than having separate data structures

Reply:
This is how we do settings throughout Zed.